src/main.ts

```diff
@@ -1,4 +1,9 @@
-export { default as createProcessor, skip } from "./process.js";
-export { pinToLocal, pinToProvider } from "./process/pinContentToIpfs.js";
-export { default as truncate } from "./process/truncate.js";
-export { default as symetricEncrypt } from "./process/symetricEncrypt.js";
+export { default as createProcessor, skip, RowObject } from "./processor.js";
+export {
+  pinToLocal,
+  pinToProvider,
+  Pinner,
```

Comment:
I didn't see an easy way to enable passing the ipfs connection details to the pinning processor, so I abstracted it into a class that get instantiated with the connection details.
This is the majority of the refactoring i ended up doing.